app/views/unattended/provisioning_templates/user_data/preseed_autoinstall_cloud_init.erb

```diff
@@ -24,11 +24,14 @@ autoinstall:
       - arches: [default]
         uri: http://ports.ubuntu.com/ubuntu-ports
 <%= indent(4) { snippet_if_exists(template_name + " custom apt") } -%>
-  identity:
-    hostname: <%= @host.name %>
-    realname: <%= realname_to_create %>
-    username: <%= username_to_create %>
-    password: "<%= password_to_create %>"
+  user-data:
+    fqdn: <%= @host.name %>
+    users:
+    - name: <%= username_to_create %>
+      gecos: <%= realname_to_create %>
+      lock-passwd: false
+      hashed_passwd: <%= password_to_create %> 
+      ssh_authorized_keys: [<%= host_param('remote_execution_ssh_keys').is_a?(String) ? host_param('remote_execution_ssh_keys') : host_param('remote_execution_ssh_keys').join("\n") %>]
```

Comment:
This will fail if `host_param('remote_execution_ssh_keys')` has not been set. Moveover, if `host_param('remote_execution_ssh_keys')` is an array it will not generate the right YAML, so I guess it'll be safer to use:
```suggestion
<% if !host_param('remote_execution_ssh_keys').blank? -%>
<% if host_param('remote_execution_ssh_keys').is_a?(String) -%>
    ssh_authorized_keys: [<%= host_param('remote_execution_ssh_keys') %>]
<% else -%>
    ssh_authorized_keys: <%= host_param('remote_execution_ssh_keys') %>
<% end -%>
<% end -%>
```